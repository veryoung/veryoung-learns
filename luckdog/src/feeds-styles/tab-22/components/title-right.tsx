import React, { ReactElement } from 'react';
import { View, Text, Image, StyleSheet } from '@tencent/hippy-react-qb';
import FeedsIcon from '../../../framework/FeedsIcon';
import { FeedsLineHeight, FeedsUIStyle } from '../../../framework/FeedsConst';
import FeedsTheme from '../../../framework/FeedsTheme';

type TitleRightProps = {
  onClick?: () => void;
  showRedDot?: boolean;
  text: string;
};

/** 红点 */
const RedDot = (): ReactElement => <View style={styles.redDot} />;

export const TitleRight = ({ onClick, showRedDot = false, text }: TitleRightProps): ReactElement => (
  <View style={styles.moreView} onClick={onClick}>
    {showRedDot ? <RedDot /> : null}
    <Text style={[styles.moreStr]} numberOfLines={1}>
      {text}
    </Text>
    <Image source={{ uri: FeedsIcon.novel_card_arrow }} style={styles.arrow} noPicMode={{ enable: false }} />
  </View>
);

const styles = StyleSheet.create({
  moreView: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
  moreStr: {
    colors: FeedsTheme.SkinColor.N1,
    fontSize: FeedsUIStyle.T1,
    lineHeight: FeedsLineHeight.T1,
  },
  arrow: {
    height: 9,
    width: 5,
    marginLeft: 4,
    tintColors: FeedsTheme.SkinColor.N1,
  },
  redDot: {
    height: 5,
    width: 5,
    borderRadius: 2.5,
    backgroundColors: FeedsTheme.LiteColor.B2,
    marginRight: 5,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
