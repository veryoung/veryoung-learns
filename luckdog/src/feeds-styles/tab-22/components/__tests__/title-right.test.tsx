import React from 'react';
import renderer from 'react-test-renderer';

import { TitleRight } from '../title-right';
import { commonProps } from '@test/mocks/props';

describe('<RankTabList />', () => {
  describe('render testing', () => {
    it('render normal', () => {
      const props = {
        ...commonProps,
        text: '查看全部',
      };
      const tree = renderer.create(<TitleRight {...props} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
