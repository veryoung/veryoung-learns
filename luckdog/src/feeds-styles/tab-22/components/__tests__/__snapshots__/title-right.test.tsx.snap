// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RankTabList /> render testing render normal 1`] = `
<div
  nativeName="View"
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-end",
    }
  }
>
  <p
    nativeName="Text"
    numberOfLines={1}
    style={
      Array [
        Object {
          "colors": Array [
            "#000000",
            "#686D74",
            "#000000",
            "#ffffff",
          ],
          "fontSize": 12,
          "lineHeight": 16,
        },
      ]
    }
    text="查看全部"
  >
    查看全部
  </p>
  <img
    alt=""
    nativeName="Image"
    noPicMode={
      Object {
        "enable": false,
      }
    }
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA8AAAAbCAYAAACjkdXHAAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAD6ADAAQAAAABAAAAGwAAAACSVupLAAAA40lEQVQ4EZ1UOQ4CMQyMQDQ8iIqWJ3HDAhXiEJf2KZS0PAX2B7QwRrFkuZo40sheecZJ1nZSSmkDNMACKF5vKL4Z2xJ1G+QWMMiiPmwXeORvysjRdXexe0plSGv4NsHBxCi3cgmOlMqQli7BycQoV8pmr3CmVIY0cwkuJka5U5fgSqkMaeIS3EyMcscuQS0q6TBmPUH6ANqJPfgdRmg5d3xoFWSY6DUEU4Vi6SEaOeH/zsy24b8drnO4w+bujnRvh6cqPM8rd1T6JQm/Yb4cO6b4ypEe1bajW07F8uS8AOmkovUDoSNiNZ1r/OEAAAAASUVORK5CYII="
    style={
      Object {
        "height": 9,
        "marginLeft": 4,
        "tintColors": Array [
          "#000000",
          "#686D74",
          "#000000",
          "#ffffff",
        ],
        "width": 5,
      }
    }
  />
</div>
`;
